driver-sync/src/main/com/mongodb/client/internal/MongoClientImpl.java

```diff
@@ -61,7 +64,14 @@ public MongoClientImpl(final Cluster cluster, final MongoClientSettings settings
                            @Nullable final OperationExecutor operationExecutor) {
         this.settings = notNull("settings", settings);
         this.delegate = new MongoClientDelegate(notNull("cluster", cluster),
-                Collections.singletonList(settings.getCredential()), this, operationExecutor);
+                singletonList(settings.getCredential()), this, operationExecutor);
+    }
+
+    public void init() {
```

Comment:
Need an init method because we have to pass a reference to `this`, and that's not safe to do in the constructor.